src/policy/compiler.rs

```diff
@@ -840,7 +840,14 @@ where
             insert_wrap!(AstElemExt::terminal(Terminal::True));
         }
         Concrete::Key(ref pk) => {
-            insert_wrap!(AstElemExt::terminal(Terminal::PkH(pk.to_pubkeyhash())));
+            insert_wrap!(AstElemExt::terminal(Terminal::PkH(
```

Comment:
Ideally this `insert_wrap` should have been avoided (since we know `Concrete::Key` during compilation), but we have a change in internal representation for `Terminal::PkH` (found when one of tests fail in absence of this statement). As I understand, there can be 2 solutions:
1. Implement `PartialEq`, `Eq` for  `Terminal::PkH` to only check equality in the field `Pk::Hash`. We can still access the `Pk` entry while having proper checks.
2. Keep both the compilations. Although I think this might give rise to ambiguity in the compilation and parsing miniscript from string can be an issue (internal representation of `Terminal::PkH` can't be inferred from a string representation as of now).